bundles/async_remote_services/rsa_spi/include/celix/rsa/IExportServiceFactory.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#pragma once
+
+#include <memory>
+
+namespace celix::rsa {
+
+    /**
+     * @brief ExportService class which represent a (opaque) exported service.
+     * The lifetime of this object should be coupled with the lifetime of the exported service.
+     */
+    class ExportedService {
+    public:
+        virtual ~ExportedService() noexcept = default;
+
+        /**
+         * @brief returns the endpoint for this exported service (to be used in discovery).
+         */
+        virtual const celix::rsa::Endpoint& getEndpoint() const = 0;
+    };
+
+    /**
+     * @brief A export service factory for a specific service type.
+     *
+     * The service type which this export service factory targets is provided with
+     * the mandatory celix::rsa::IExportServiceFactory::TARGET_SERVICE_NAME service property.
+     *
+     */
+    class IExportServiceFactory {
```

Comment:
The impact of switching over to an opaque factory (either imported or exported) I think requires some more thought. Some of the impact that I can foresee right now is:
* Factory implementations would add service dependencies instead of passing the `void*` directly, for each service created.
* This would mean that the `exportService` function below should not return the exportedService. The admin should add a tracker for that as well.
* Factories should probably also have the new services have a dependency on the factory itself.  This and the previous point would solve the problem with creating/destroying bundles IMO

Reply:
- Correct. The factory knows the type and can get the svc id from the service properties, as result it can create a ExportedService which a service dependency to the service. 

- If I am correct the current setup is that an exported service is marked using a IExportedService interface. So the ExportServiceFactory creates a ExportService which is a component with a publish, subscriber, dependency on the "service to export" and a IExportedService provided interface. 
The IExportedService services can be tracked by the topology manager or discovery manager so that it can be announced in the network. The async admin does not need to know this, it only has to keep the std::unique_ptr<ExportService> as long as the "service to export" and the Factory which created the ExportService is tracked/available. 
Note that for now the "announce in network" is not relevant; configured discovery only handles imports. 

- Yes, exactly. And an IMO additional benefit is that the 'dm' overview will also cleary show which service is exported by a ExportService component.